frontend/src/metabase/query_builder/components/filters/FilterPopover.tsx

```diff
@@ -114,17 +115,19 @@ export default class FilterPopover extends Component<Props, State> {
     );
   };
 
-  handleCommitFilter = (filter: Filter | null, query: StructuredQuery) => {
-    if (filter && !(filter instanceof Filter)) {
-      filter = new Filter(filter, null, query);
-    }
-    if (filter && filter.isValid() && this.props.onChangeFilter) {
-      this.props.onChangeFilter(filter);
-      if (this.props.onClose) {
-        this.props.onClose();
+  handleCommitFilter = _.once(
```

Comment:
At first, I don't like `_.once` because it feels like slapping a patch without addressing the actual issue. But no matter how much I tried, I couldn't come up with any better way.

Maybe the only concern is that when time passes we'll lose context why this function needs to be only called once. What if you add a comment and link to this issue, so future readers would understand the intent behind this change.
e.g. 
https://github.com/metabase/metabase/blob/3b8d88b5ab2ce8a45f8aab0b11776d76dc900603/frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx#L16-L19